runners/flink/src/main/java/org/apache/beam/runners/flink/metrics/FlinkMetricContainer.java

```diff
@@ -17,299 +17,43 @@
 public class FlinkMetricContainer {
 
-  private static final String METRIC_KEY_SEPARATOR = "__";
-  static final String COUNTER_PREFIX = "__counter";
-  static final String DISTRIBUTION_PREFIX = "__distribution";
-  static final String GAUGE_PREFIX = "__gauge";
+  private static final Logger LOG = LoggerFactory.getLogger(FlinkMetricContainer.class);
 
-  private final MetricsContainer metricsContainer;
-  private final RuntimeContext runtimeContext;
-  private final Map<String, Counter> flinkCounterCache;
-  private final Map<String, FlinkDistributionGauge> flinkDistributionGaugeCache;
-  private final Map<String, FlinkGauge> flinkGaugeCache;
+  public static final String ACCUMULATOR_NAME = "__metricscontainers";
 
-  public FlinkMetricContainer(String stepName, RuntimeContext runtimeContext) {
-    metricsContainer = new MetricsContainer(stepName);
-    this.runtimeContext = runtimeContext;
-    flinkCounterCache = new HashMap<>();
-    flinkDistributionGaugeCache = new HashMap<>();
-    flinkGaugeCache = new HashMap<>();
-  }
-
-  public MetricsContainer getMetricsContainer() {
-    return metricsContainer;
-  }
-
-  public void updateMetrics() {
-    // update metrics
-    MetricUpdates updates = metricsContainer.getUpdates();
-    if (updates != null) {
-      updateCounters(updates.counterUpdates());
-      updateDistributions(updates.distributionUpdates());
-      updateGauge(updates.gaugeUpdates());
-      metricsContainer.commitUpdates();
-    }
-  }
-
-  private void updateCounters(Iterable<MetricUpdates.MetricUpdate<Long>> updates) {
-
-    for (MetricUpdates.MetricUpdate<Long> metricUpdate : updates) {
-
-      String flinkMetricName = getFlinkMetricNameString(COUNTER_PREFIX, metricUpdate.getKey());
-      Long update = metricUpdate.getUpdate();
-
-      // update flink metric
-      Counter counter = flinkCounterCache.get(flinkMetricName);
-      if (counter == null) {
-        counter = runtimeContext.getMetricGroup().counter(flinkMetricName);
-        flinkCounterCache.put(flinkMetricName, counter);
-      }
-      counter.dec(counter.getCount());
-      counter.inc(update);
-
-      // update flink accumulator
-      Accumulator<Long, Long> accumulator = runtimeContext.getAccumulator(flinkMetricName);
-      if (accumulator == null) {
-        accumulator = new LongCounter(update);
-        runtimeContext.addAccumulator(flinkMetricName, accumulator);
-      } else {
-        accumulator.resetLocal();
-        accumulator.add(update);
-      }
-    }
-  }
-
-  private void updateDistributions(Iterable<MetricUpdates.MetricUpdate<DistributionData>> updates) {
-
-    for (MetricUpdates.MetricUpdate<DistributionData> metricUpdate : updates) {
-
-      String flinkMetricName =
-          getFlinkMetricNameString(DISTRIBUTION_PREFIX, metricUpdate.getKey());
-      DistributionData update = metricUpdate.getUpdate();
-
-      // update flink metric
-      FlinkDistributionGauge gauge = flinkDistributionGaugeCache.get(flinkMetricName);
```

Comment:
It looks like this is removing the support for exporting Beam Metrics to Flink Metrics. (Same holds for `updateCounters()` and `updateGauge()`. 

Unfortunately this is not tested right now since this is somewhat hard to do. I created a Jira issue for this: https://issues.apache.org/jira/browse/BEAM-2056